### DISCUSSION TITLE
Update React-Hook-Form website Typescript Version Recommendation

### DISCUSSION THREAD

**Discussion by Jamus123** [react-hook-form/react-hook-form#11869] [Ideas]
Currently the website suggest TS version ^4.3. However, I just spent the better part of a day figuring out a 'Type Instantiation is excessively deep and possibly infinite' error while on 4.4 on using the FormProvider component. Was on 7.51.4.

Small suggestion to show TS to react-hook-form version matching. In other words, just show a list of TS you should use vs the version of react-hook-form you are on.

**Comment by avinoamsn** [comment]
@Jamus123 can you share the version of TS that resolved the error, in addition to the more details about the type definitions in question?

The TS instantiation depth is somewhat arbitrary and dependent on version-specific implementation details (per https://github.com/microsoft/TypeScript/pull/44997#issuecomment-900595879), so it's likely that what works in your situation will not apply universally. Nevertheless, it would be good to document this information for posterity.
